torch/testing/_internal/common_nn.py

```diff
@@ -4905,120 +4989,124 @@ def _do_test(self, test_case, module, input):
 
             module_ip = self.constructor(*self.constructor_args, inplace=True)
 
-            input_version = input._version
+            input_versions = tuple(t._version for t in input_tuple)
             with freeze_rng_state():
-                output = module(input)
-            test_case.assertEqual(input._version, input_version)
+                output = module(*input_tuple)
+            new_input_versions = tuple(t._version for t in input_tuple)
+            test_case.assertEqual(input_versions, new_input_versions)
 
-            input_ip = deepcopy(input)
-            input_ip_clone = input_ip.clone()
+            input_ip = deepcopy(input_tuple)
+            input_ip_clone = tuple(t.clone() for t in input_ip)
             with freeze_rng_state():
-                output_ip = module_ip(input_ip_clone)
-            test_case.assertNotEqual(input_ip_clone._version, input_version)
+                output_ip = module_ip(*input_ip_clone)
+            input_ip_clone_versions = tuple(t._version for t in input_ip_clone)
+            test_case.assertNotEqual(input_ip_clone_versions, input_versions)
             test_case.assertEqual(output, output_ip)
             grad = output.data.clone().normal_()
-            input.grad.data.zero_()
+            for t in input_tuple:
+                t.grad.data.zero_()
             output.backward(grad)
             output_ip.backward(grad)
-            test_case.assertEqual(input.grad, input_ip.grad)
+            for t1, t2 in zip(input_tuple, input_ip):
+                test_case.assertEqual(t1.grad, t2.grad)
 
-        if isinstance(input, torch.LongTensor) and TEST_CUDA:
+        if any(isinstance(t, torch.LongTensor) for t in input_tuple) and TEST_CUDA:
             # check that cuda() moves module parameters to correct GPU device,
             # and that float() casts parameters correctly
-
-            input = input.cuda()
+            input_tuple = tuple(t.cuda() for t in input_tuple)
```

Comment:
Can we just assert that `len(input_tuple) = 1` and use `input = input_tuple[0]` instead?

It doesn't quite make sense to me that an in-place version of a module should take in multiple tensors, because there is a confusion over "which of the multiple tensor arguments is actually being operated on in-place?". Furthermore Transformer has no in-place variant

Reply:
@zou3519 
I made this change to allow tuple inputs everywhere, not for Transformer only.
Do you mean we don't have the following cases?
1. Module.forward(_0, _1, _2...),  forward has multiple inputs and all of them are inplace var.
2. Module.forwar(_0, _1, _2...), forward has multiple inputs, and some of them are inplace var, eg (_0 and _2 are inplace)

The inplace check now can handle all above cases, I am not sure whether this is needed. If not, I will revert the changes to inplace_check and assert len(input_tuple) == 1.

Reply:
All of the inplace Modules that I am aware of (see [search](https://pytorch.org/docs/stable/search.html?q=inplace&check_keywords=yes&area=default), and by grepping common_nn.py for `check_inplace`) take a single Tensor as an input and modify it in-place. These consist of activation functions and dropout. I don't think we have any in-place modules that take in multiple tensors in the forward pass.

Reply:
I see. I will do `assert len(input_tuple) == 1` and leave a comment there.